### PR TITLE
Change to Vite and Add Typescript Support

### PR DESCRIPTION
**Pull request by hisyamnasir**
1. Change Create React App (CRA) to React Vite
2. Add Typescript support
3. Add eslint and prettier

For production deployment change inside `dashboard/.env` file from `development` to `production`

```env
REACT_APP_ENV=development
```
to
```env
REACT_APP_ENV=production
```

### CODE REVIEW

File: dashboard/package.json [hunk 1]
```diff
@@ -5,14 +5,16 @@
...
   "scripts": {
+    "dev": "npm run start",
     "build": "echo \"Please use build:dev or build:prod \" && exit 1",
-    "build:dev": "cross-env REACT_APP_ENV=development react-scripts build",
-    "build:prod": "cross-env REACT_APP_ENV=production react-scripts build",
-    "eject": "react-scripts eject",
-    "lint": "eslint \"src/**/*.js\"",
-    "start": "cross-env REACT_APP_ENV=development react-scripts start",
-    "start:prod": "cross-env REACT_APP_ENV=production react-scripts start",
+    "build:dev": "cross-env-shell REACT_APP_ENV=development tsc && vite build",
```
[the remark this answers is not in the document]
Comment (hisyamnasir) [reply dashboard/package.json:12]: ok, done fix.

Comment (rohaizanr) [inline dashboard/package.json:12]: @hisyamnasir ```npm run build:dev``` yang baru ni akan break ```sh run-dev.sh```

1. Mungkin build folder dah di tukar ke dist folder
2. Edit Dockerfile.dev sekali untuk refer folder yang baru
3. Make sure ```sh run-dev.sh``` & http://localhost:3000 is working

**Review by rohaizanr — CHANGES_REQUESTED** [review]

File: dashboard/package.json (hunk 1 above)
Comment (hisyamnasir) [reply dashboard/package.json:12]: Fix, change to dist folder.

**Review by rohaizanr — APPROVED** [review]
awesome work! 
terima kasih

**Pull request merged**